src/simple_server/core.clj

```diff
@@ -14,21 +14,68 @@
 ;;; and our "http code".  Our game now lives in its own namespace, and
 ;;; is fully testable independent of our "presentation layer".
 
-(defn new-game-handler []
-  (when (game/new-game!)
-    (response "OK- start guessing at /guess")))
-
-(defn guess-handler [guess]
-  (condp = (game/guess-answer guess)
-    nil       (-> (response  "You need to supply a guess with /guess?guess=N")
-                  (status 400))
-    :game-over (response  "Congratulations! You win!")
-    :too-low   (response "Too low.")
-    :too-high  (response  "Too high.")))
+;; An abstraction for accounts in the database
+(def users (atom #{"ray"}))
+
+;; An abstraction for currently logged in accounts
+(def current-users (atom #{}))
+
+(defn landing-handler []
+  ;; The landing page of the game, with instructions to tell user to either sign up or log in
+  (response (str "Welcome to Guessing the Number! "
+                 "Head to /sign-up?account=\"your account\" to sign up if this is your first time visiting. "
+                 "Or go to /log-in?account=\"your account\" to log in to start playing the game!")))
+
+(defn sign-up-handler [account]
+  ;; Check if the user already exists, if not, create it, otherwise, tell the user to log in
+  (if (nil? (some #{account} @users))
+    (do
+      (swap! users conj account)
+      (response (str "Thank you for joining us! " account ". You may now log in to your page at /login?account=" account)))
+    (response (str  "You are already with us " account ", log in to your page at /login?account=" account))))
+
+(defn log-in-handler [account]
+  ;; Check if the user is in our database, if not, tell the user to sign up, otherwise, log the user in
+  (if (nil? (some #{account} @users))
+    (response "You are not a member of us yet, pleaes sign up at /sign-up?account=\"your account\".")
+    (do
+      (swap! current-users conj account)
+      (response (str "Welcome back " account ", head to your page at /" account)))))
+
+(defn account-handler [account]
+  ;; Check if the user is logged in, if not, tell the user to log in, otherwise, welcome the user
+  (if (nil? (some #{account} @current-users))
```

Comment:
This pattern, `(nil? (some #{account} @current-users))`, sounds like `(logged-in? account)` to me.

Reply:
You are right, I should have created a function for that